src/Nethermind/Nethermind.KeyStore/PasswordProvider.cs

```diff
@@ -0,0 +1,107 @@
+//  Copyright (c) 2018 Demerzel Solutions Limited
+//  This file is part of the Nethermind library.
+// 
+//  The Nethermind library is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+// 
+//  The Nethermind library is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//  GNU Lesser General Public License for more details.
+// 
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+// 
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Security;
+using Nethermind.Crypto;
+using Nethermind.KeyStore.Config;
+using Nethermind.Logging;
+
+namespace Nethermind.KeyStore
+{
+    public class PasswordProvider : IPasswordProvider
+    {
+        private readonly IKeyStoreConfig _keyStoreConfig;
+        public PasswordProvider(IKeyStoreConfig keyStoreConfig)
+        {
+            _keyStoreConfig = keyStoreConfig ?? throw new ArgumentNullException(nameof(keyStoreConfig));
+        }
+        public SecureString GetBlockAuthorPassword()
+        {
+            SecureString passwordFromFile = null;
+            var index = Array.IndexOf(_keyStoreConfig.UnlockAccounts, _keyStoreConfig.BlockAuthorAccount);
+            if (index >= 0)
+            {
+                passwordFromFile = GetPassword(index);
+            }
+
+            return passwordFromFile != null ? passwordFromFile : GetPasswordFromConsole();
+        }
+
+        public SecureString GetPassword(int keyStoreConfigPasswordIndex)
+        {
+            string GetPasswordN(int n, string[] passwordsCollection) => passwordsCollection?.Length > 0 ? passwordsCollection[Math.Min(n, passwordsCollection.Length - 1)] : null;
+
+            SecureString password = null;
+            var passwordFile = GetPasswordN(keyStoreConfigPasswordIndex, _keyStoreConfig.PasswordFiles);
+            if (passwordFile != null)
+            {
+                string blockAuthorPasswordFilePath = passwordFile.GetApplicationResourcePath();
+                password = File.Exists(blockAuthorPasswordFilePath)
```

Comment:
in original it was  ove IFileSystem

Reply:
I know. However, it was used for mocking in tests. Now, we did it in another way, and we don't need it in tests anymore. Thanks to that, we reduced the number of injected dependencies. If you think that we should return it, I will do it.